### PR TITLE
Karinamzalez bcda 6023

### PR DESCRIPTION
**Pull request by karinamzalez**
## 🎫 Ticket

https://jira.cms.gov/browse/BCDA-6023

## 🛠 Changes

- Adds Expired Token error handling 

## ℹ️ Context for reviewers

(Background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers.)

- For an expired token scenario, a 401 is returned and we are now displaying "Expired Token" as the detailsCode value displayed in the FHIR operational outcome to be more explicit. 

## ✅ Acceptance Validation

(How were the changes verified? Did you fully test the acceptance criteria in the ticket? Provide reproducible testing instructions and screenshots if applicable.)

- Updated unit test for token verification error handling in middleware_test.go > TestTokenVerificationErrorHandling 
- Added integration test that tests API call with an expired jwt token in middleware_test.go > TestReturn401WhenExpiredToken

## 🔒 Security Implications

- [ ] New data is stored or transmitted.
  

- [ ] This change introduces new software dependencies.
  

- [ ] This change impacts security controls or alters supporting software.
  

- [ ] A [security checklist](https://confluence.cms.gov/display/BCDA/Security+Checklists) is completed for this change.
  

- [ ] More team discussion required to evaluate security implications.

### CODE REVIEW

File: bcda/auth/middleware_test.go [hunk 1]
```diff
@@ -80,7 +81,34 @@ func (s *MiddlewareTestSuite) TestReturn400WhenInvalidTokenAuthWithInvalidSignat
...
+
+	req, err := http.NewRequest("GET", fmt.Sprintf(constants.ServerPath, s.server.URL), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf(bearerStringMsg, expiredToken))
+	resp, err := client.Do(req)
+
+	assert.NotNil(s.T(), resp)
```
Comment (laurenkrugen-navapbc) [inline bcda/auth/middleware_test.go:105]: This looks great! I'm not sure I see it here- are we also testing for this acceptance criteria in the ticket? 

`the detailsCode in FHIR Operational Outcome (response body) is "Expired Token"`

Comment (karinamzalez) [reply bcda/auth/middleware_test.go:105]: @laurenkrugen-navapbc yes!

Comment (laurenkrugen-navapbc) [reply bcda/auth/middleware_test.go:105]: sorry - I think my eyes are missing it; I see the assertion that the resp is not not nil and that we are getting the correct status code, but I don't see a comparison for the response body.. can you point me to the line or give me a quick walkthrough?

Comment (karinamzalez) [reply bcda/auth/middleware_test.go:109]: @laurenkrugen-navapbc up to date!

Comment (karinamzalez) [reply bcda/auth/middleware_test.go:109]: Thank you for catching this :)

**Review by laurenkrugen-navapbc — APPROVED** [review]
LGTM!

**Pull request merged**